### PR TITLE
Vite: Add web-components/lit framework support

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

- [x] web-components/vite framework
- [x] JS/TS repro-templates
- [x] web-components improvements
  - [x] default render function 🎉 
  - [x] template components
  - [x] template stories
- [ ] delete web-components-kitchen-sink

## How to test

- [ ] CI passes

### CODE REVIEW

**Comment by shilman** [comment]
@ndelangen the repro generation is failing because it uses `yarn bootstrap --prep` and that fails on the angular examples with errors like this:

```
> nx run @storybook/angular:prep --reset

ERR! FAILED (ts) : ../../lib/api/src/modules/versions.ts:2:1 - error TS2578: Unused '@ts-expect-error' directive.

2 // @ts-expect-error no typedefs
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The same commands succeed in production mode, so hopefully we can fix the inconsistency somehow.

**Review by tmeasday — APPROVED** [review]
LGTM.

> Maybe this should be lit-vite and generate lit stories?

Do you mean the CLI template should be lit stories? I have no idea, I'm not the right person to ask this!

File: code/lib/cli/src/repro-templates.ts [hunk 1]
```diff
@@ -94,11 +94,35 @@ const svelteViteTemplates = {
   // }
 };
 
+const webComponentsViteTemplates = {
+  'web-components-vite/default-js': {
```
Comment (tmeasday) [inline code/lib/cli/src/repro-templates.ts:98]: This should be `lit-vite/default-js`

File: code/addons/interactions/template/stories/basics.stories.ts [hunk 2]
```diff
@@ -46,23 +51,30 @@ export const Callback = {
 // an explicit test here
 export const SyncWaitFor = {
   play: async ({ canvasElement, step }) => {
-    const canvas = within(canvasElement);
+    const canvas = safeWithin(canvasElement);
     await step('Setup', Callback.play);
-    await waitFor(() => canvas.getByText('Completed!!'));
+
+    // FIXME: why doesn't this work?
```
Comment (shilman) [inline code/addons/interactions/template/stories/basics.stories.ts:57]: @tmeasday I had to change this to get web-components to work due to some shadowRoot-related issue that I don't understand. Is this ok?

Comment (tmeasday) [reply code/addons/interactions/template/stories/basics.stories.ts]: We took a look at this and (semi-) resolved it. It was due to bundling different versions (`.js` + `.mjs`) of the instrumenter somehow.

**Pull request merged**